tso-server/src/main/java/com/yahoo/omid/tso/PersistenceProcessorHandler.java

```diff
@@ -0,0 +1,274 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yahoo.omid.tso;
+
+import static com.codahale.metrics.MetricRegistry.name;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+import org.jboss.netty.channel.Channel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.lmax.disruptor.WorkHandler;
+import com.yahoo.omid.committable.CommitTable;
+import com.yahoo.omid.metrics.Histogram;
+import com.yahoo.omid.metrics.MetricsRegistry;
+import com.yahoo.omid.metrics.Timer;
+import com.yahoo.omid.tso.BatchPool.Batch;
+import com.yahoo.omid.tso.PersistEvent.Type;
+
+    public class PersistenceProcessorHandler
+    implements
+    WorkHandler<PersistenceProcessorImpl.PersistBatchEvent> {
+
+        private static final Logger LOG = LoggerFactory.getLogger(PersistenceProcessor.class);
+
+        private final String tsoHostAndPort;
+        private final LeaseManagement leaseManager;
+
+        final ReplyProcessor reply;
+        final RetryProcessor retryProc;
+        final CommitTable.Client commitTableClient;
+        final CommitTable.Writer writer;
+        final Panicker panicker;
+
+        final int maxBatchSize;
+
+        Batch handlerBatch;
+
+        final Timer flushTimer;
+        final Histogram batchSizeHistogram;
+
+        long lastFlush;
+
+        PersistenceProcessorHandler(MetricsRegistry metrics,
+                String tsoHostAndPort,
+                LeaseManagement leaseManager,
+                CommitTable commitTable,
+                ReplyProcessor reply,
+                RetryProcessor retryProc,
+                Panicker panicker,
+                TSOServerConfig config)
+                        throws InterruptedException, ExecutionException, IOException {
+
+            this.tsoHostAndPort = tsoHostAndPort;
+            this.leaseManager = leaseManager;
+            this.commitTableClient = commitTable.getClient();
+            this.writer = commitTable.getWriter();
+            this.reply = reply;
+            this.retryProc = retryProc;
+            this.panicker = panicker;
+            this.maxBatchSize = config.getMaxBatchSize() / config.getPersistHandlerNum();
+
+            LOG.info("Creating the persist processor with batch size {}, and timeout {}ms",
+                    maxBatchSize, config.getBatchPersistTimeoutMS());
+
+            flushTimer = metrics.timer(name("tso", "persist", "flush"));
+            batchSizeHistogram = metrics.histogram(name("tso", "persist", "batchsize"));
+        }
+
+        @Override
+        public void onEvent(PersistenceProcessorImpl.PersistBatchEvent event) throws Exception {
+            Batch batch = event.getBatch();
+            handlerBatch = batch;
```

Comment:
handlerBatch attribute is not required. You should remove it and pass the batch to the flush() method:

```
private void flush(Batch batch, long batchID) {
...
}
```